crates/containerd-shim-wasm/src/sandbox/sys/windows/stdio.rs

```diff
@@ -0,0 +1,42 @@
+use std::io::{Error, ErrorKind, Result};
+use std::os::windows::prelude::{AsRawHandle, IntoRawHandle, OwnedHandle};
+
+type RawFd = libc::c_int;
+
+pub static STDIN_FILENO: RawFd = 0;
+pub static STDOUT_FILENO: RawFd = 1;
+pub static STDERR_FILENO: RawFd = 2;
+
+struct OwnedFd {
+    fd: RawFd,
+}
+
+pub fn try_into_fd(f: impl Into<OwnedHandle>) -> Result<impl AsRawFd> {
+    let handle = f.into();
+    let raw = handle.as_raw_handle() as libc::intptr_t;
+    let fd = unsafe { libc::open_osfhandle(raw, libc::O_APPEND) };
+    if fd == -1 {
+        return Err(Error::new(
+            ErrorKind::Other,
+            "Failed to open file descriptor",
+        ));
+    }
+    let _ = handle.into_raw_handle(); // drop ownership of the handle, it's managed by fd now
+    Ok(OwnedFd { fd })
+}
+
+pub trait AsRawFd {
```

Comment:
I am not sure how I feel about implementing `AsRawFd` and related for windows.  I haven't really seen this pattern in other projects.  (though my experience is limited tbh). 

This seems fairly contained to use inside the stdio package so I think this should be ok

Reply:
I know, I thought it would be fine since it's contained to the module.
I was being lazy, so I would save writing a trait in the unix case.
But we could use a different name as well :-)

Reply:
Ah, as long as the method name `as_raw_fd` doesn't change, it should be fine.